File: backend/app/schemas/restoration.py
"""文物修复相关 Pydantic Schema"""

from datetime import datetime
from pydantic import BaseModel


class DamageAnalysis(BaseModel):
    """Step 1 输出: 损伤分析"""
    category: str
    damage_types: list[str]
    severity: str
    description: str


class RestorationPromptResult(BaseModel):
    """Step 2 输出: 修复方案"""
    prompt: str


class ImageRestorationResult(BaseModel):
    """Step 3 输出: 图像修复"""
    images: list[str]
    seed: int


class VerificationDimensions(BaseModel):
    """Step 4 评估维度"""
    detail_fidelity: int = 0       # 细节保真度
    style_consistency: int = 0     # 风格一致性
    restoration_completeness: int = 0  # 损伤修复完整度


class VerificationReport(BaseModel):
    """Step 4 输出: 修复验证"""
    overall_score: int
    dimensions: VerificationDimensions
    verdict: str
    artifacts: list[str] = []


class PipelineStep(BaseModel):
    """管道步骤"""
    step: int
    name: str
    model: str
    status: str  # completed / failed
    result: dict | None = None


class RestorationResponse(BaseModel):
    """POST /upload 响应"""
    id: int
    original_image_url: str
    restored_image_url: str | None = None
    pipeline_steps: list[PipelineStep]
    created_at: datetime

    model_config = {"from_attributes": True}


class RestorationListItem(BaseModel):
    """GET /history 列表项"""
    id: int
    original_image_url: str
    restored_image_url: str | None = None
    damage_category: str | None = None
    verification_score: int | None = None
    pipeline_status: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/services/restoration.ts
import api from './api'

// ============================================================
// Types
// ============================================================

export interface DamageAnalysis {
  category: string
  damage_types: string[]
  severity: string
  description: string
}

export interface RestorationPromptResult {
  prompt: string
}

export interface ImageRestorationResult {
  images: string[]
  seed: number
}

export interface VerificationDimensions {
  detail_fidelity: number
  style_consistency: number
  restoration_completeness: number
}

export interface VerificationReport {
  overall_score: number
  dimensions: VerificationDimensions
  verdict: string
  artifacts: string[]
}

export interface PipelineStep {
  step: number
  name: string
  model: string
  status: string // 'completed' | 'failed'
  result: Record<string, any> | null
}

export interface RestorationResult {
  id: number
  original_image_url: string
  restored_image_url: string | null
  pipeline_steps: PipelineStep[]
  created_at: string
}

export interface RestorationListItem {
  id: number
  original_image_url: string
  restored_image_url: string | null
  damage_category: string | null
  verification_score: number | null
  pipeline_status: string
  created_at: string
}

export interface PaginatedResponse<T> {
  items: T[]
  total: number
  page: number
  pages: number
}

// ============================================================
// API Functions
// ============================================================

export async function uploadAndRestore(file: File): Promise<RestorationResult> {
  const formData = new FormData()
  formData.append('file', file)
  const res = await api.post('/api/restoration/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
    timeout: 300000, // 5min — pipeline takes time
  })
  return res.data
}

export async function getHistory(page = 1, pageSize = 10): Promise<PaginatedResponse<RestorationListItem>> {
  const res = await api.get('/api/restoration/history', {
    params: { page, page_size: pageSize },
  })
  return res.data
}

export async function getDetail(id: number): Promise<RestorationResult> {
  const res = await api.get(`/api/restoration/${id}`)
  return res.data
}

export async function deleteRestoration(id: number) {
  const res = await api.delete(`/api/restoration/${id}`)
  return res.data
}
